quantum/pointing_device/pointing_device_modes.h

```diff
@@ -0,0 +1,167 @@
+/* Copyright 2023 Alabastard (@Alabastard-64)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <string.h>
+#include "quantum.h"
+#include "pointing_device.h"
+
+// default settings
+#ifndef POINTING_TAP_DELAY
+#    define POINTING_TAP_DELAY TAP_CODE_DELAY
+#endif
+#ifndef POINTING_MODE_DEFAULT
+#    define POINTING_MODE_DEFAULT PM_NONE
+#endif
+
+#if defined(SPLIT_POINTING_ENABLE) && defined(POINTING_DEVICE_COMBINED)
+#    ifndef POINTING_MODES_NUM_DEVICES
+#        define POINTING_MODES_NUM_DEVICES 2
+#    endif
+#else
+#    ifndef POINTING_MODES_NUM_DEVICES
+#        define POINTING_MODES_NUM_DEVICES 1
+#    endif
+#endif
+#ifdef POINTING_MODES_SINGLE_CONTROL
+#    define POINTING_MODES_DEVICE_CONTROL_COUNT 1
+#else
+#    define POINTING_MODES_DEVICE_CONTROL_COUNT POINTING_MODES_NUM_DEVICES
+#endif
+
+#ifndef POINTING_MODES_DEFAULT_DEVICE
+#    define POINTING_MODES_DEFAULT_DEVICE 0
+#endif
+
+// default divisors
+#ifndef POINTING_DEFAULT_DIVISOR
+#    define POINTING_DEFAULT_DIVISOR 64
+#endif
+#ifndef POINTING_HISTORY_DIVISOR
+#    define POINTING_HISTORY_DIVISOR 64
+#endif
+#ifndef POINTING_VOLUME_DIVISOR
+#    define POINTING_VOLUME_DIVISOR 64
+#endif
+#ifndef POINTING_CARET_DIVISOR
+#    define POINTING_CARET_DIVISOR 32
+#endif
+#ifndef POINTING_CARET_DIVISOR_H
+#    define POINTING_CARET_DIVISOR_H POINTING_CARET_DIVISOR
+#endif
+#ifndef POINTING_CARET_DIVISOR_V
+#    define POINTING_CARET_DIVISOR_V POINTING_CARET_DIVISOR
+#endif
+#ifndef POINTING_PRECISION_DIVISOR
+#    define POINTING_PRECISION_DIVISOR 2
+#endif
+#ifndef POINTING_DRAG_DIVISOR
+#    define POINTING_DRAG_DIVISOR 4
+#endif
+
+// error checking
+#if POINTING_DEFAULT_DIVISOR < 1
+#    pragma message "DEFAULT_DIVISOR must be 1 or greater"
+#    error DEFAULT_DIVISOR less than 1
+#endif
+#if POINTING_MODES_NUM_DEVICES < 2 && (defined(SPLIT_POINTING_ENABLE) && defined(POINTING_DEVICE_COMBINED)) || defined(POINTING_MODES_SINGLE_CONTROL)
+#    pragma message "POINTING_MODE_NUM_DEVICES should be at least 2 with SPLIT_POINTING_ENABLE & POINTING_DEVICE_COMBINED or POINTING_MODES_SINGLE_CONTROL defined"
+#    error POINTING_MODE_NUM_DEVICES set too low
+#endif
```

Comment:
Testing single device control now. I might not understanding it fully but even with `#define POINTING_MODES_NUM_DEVICES 2`, this wouldn't compile for me. So I changed it to` #if POINTING_MODES_NUM_DEVICES < 2 && defined(SPLIT_POINTING_ENABLE) && (defined(POINTING_DEVICE_COMBINED) || defined(POINTING_MODES_SINGLE_CONTROL))` and got it to compile.

Reply:
Thanks for catching all of these errors, sorry there have been so many I was rushing to get this out before a work trip.
I'll fix this up and will push a commit with fixes.